lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1926,9 +1981,10 @@ static bool processInternalGlobal(
 
     // Otherwise, if the global was not a boolean, we can shrink it to be a
     // boolean.
+    // XXX EMSCRIPTEN - add range metadata instead
     if (Constant *SOVConstant = dyn_cast<Constant>(GS.StoredOnceValue)) {
       if (GS.Ordering == AtomicOrdering::NotAtomic) {
-        if (TryToShrinkGlobalToBoolean(GV, SOVConstant)) {
+        if (TryToAddRangeMetadata(GV, SOVConstant)) { // XXX EMSCRIPTEN
```

Comment:
I think this annoyed us because it turns globals into booleans which ended up adding overhead for us (masking, etc.) on some benchmark. I doubt it makes much different natively. But as in the others, might make sense to condition it on the target being emscripten I guess.